homeassistant/config.py

```diff
@@ -1590,23 +1593,52 @@ async def async_process_component_config(  # noqa: C901
             config_exceptions.append(exc_info)
             continue
 
-        try:
-            platform = await p_integration.async_get_platform(domain)
-        except LOAD_EXCEPTIONS as exc:
-            exc_info = ConfigExceptionInfo(
-                exc,
-                ConfigErrorTranslationKey.PLATFORM_COMPONENT_LOAD_EXC,
-                platform_path,
-                p_config,
-                integration_docs,
-            )
-            config_exceptions.append(exc_info)
-            continue
+        platforms_to_load.append((p_name, p_integration, p_config))
+
+    #
+    # Since bootstrap will order base platform (ie sensor) integrations
+    # first, we eagerly gather importing the platforms that need to be
+    # validated for the base platform since everything that uses the
+    # base platform has to wait for it to finish.
+    #
+    # For example if `hue` where to load first and than called
+    # `async_forward_entry_setup` for the `sensor` platform it would have to
+    # wait for the sensor platform to finish loading before it could continue.
+    # Since the base `sensor` platform must also import all of its platform
+    # integrations to do validation before it can finish setup, its important
+    # that the platform integrations are imported first so we do not waste
+    # time importing `hue` first when we could have been importing the platforms
+    # that the base `sensor` platform need to load to do validation and allow
+    # all integrations that need the base `sensor` platform to proceed with setup.
+    #
+    if platforms_to_load:
+
+        async def _async_load_and_validate_platform(
```

Comment:
Can we move this function out?

Reply:
Yes we can.  It has a giant amount of args though so I haven't moved it yet as I'm still testing

Reply:
I ended up making a dataclass to pass it around because it was too confusing otherwise